### PR TITLE
chore: Additional Transaction component QA

### PR DESCRIPTION
**Pull request by abcrane123**
**What changed? Why?**
- hide sponsor component if user hasn't provided paymaster url
- fix issue where request denied error is absorbed
- add onsuccess handler
- add disable functionality to button

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

File: site/docs/pages/transaction/types.mdx [hunk 1]
```diff
@@ -50,6 +54,7 @@ type TransactionProviderReact = {
   children: ReactNode; // The child components to be rendered within the provider component.
   contracts: ContractFunctionParameters[]; // An array of contract function parameters provided to the child components.
   onError?: (e: TransactionError) => void; // An optional callback function that handles errors within the provider.
+  onSuccess?: (response: TransactionResponse) => void; // An optional callback function that exposes transaction hash
```
Comment (Zizzamia) [inline site/docs/pages/transaction/types.mdx:57]: Oh nice, thank you for fixing the docs.

File: site/docs/pages/transaction/types.mdx [hunk 2]
```diff
@@ -62,6 +67,16 @@ type TransactionReact = {
...
   contracts: ContractFunctionParameters[]; // An array of contract function parameters for the transaction.
   onError?: (e: TransactionError) => void; // An optional callback function that handles transaction errors.
+  onSuccess?: (response: TransactionResponse) => void; // An optional callback function that exposes transaction hash
+};
+```
+
+## `TransactionResponse`
+
+```ts
+type TransactionResponse = {
```
Comment (Zizzamia) [inline site/docs/pages/transaction/types.mdx:77]: Remember to export they types we document.

File: src/transaction/components/TransactionProvider.tsx [hunk 3]
```diff
@@ -104,6 +112,11 @@ export function TransactionProvider({
         } catch (_err) {
           setErrorMessage(genericErrorMessage);
         }
+      } else if (
```
Comment (Zizzamia) [inline src/transaction/components/TransactionProvider.tsx:115]: Can we add a line of comment for those conditions, so we don't forget for what are for.

File: site/docs/pages/transaction/types.mdx (hunk 2 above)
Comment (abcrane123) [reply site/docs/pages/transaction/types.mdx:77]: this is in the docs though. we don't export here do we ?

File: src/transaction/components/TransactionToast.tsx [hunk 4]
```diff
@@ -36,6 +36,22 @@ export function TransactionToast({
...
   }, [position]);
 
+  useEffect(() => {
+    let timer: ReturnType<typeof setTimeout> | undefined;
+    // hide toast after 5 seconds once
+    // it reaches final state (success or error)
+    if (receipt || errorMessage) {
+      timer = setTimeout(() => {
+        setIsToastVisible(false);
+      }, 5000);
```
Comment (Zizzamia) [inline src/transaction/components/TransactionToast.tsx:46]: I feel like, this is something folks want to customize at some point. Let's create a P3 Linear ticker for this. @cpcramer this is a great ticket for you, as P3, if you don't mind.

File: site/docs/components/TransactionWrapper.tsx [hunk 5]
```diff
@@ -46,10 +46,14 @@ export default function TransactionWrapper({
     console.error('TransactionWrapper:', error);
   }
 
+  function onSuccess(response: any) {
+    console.log('TransactionWrapperSuccessHandler', response)
```
Comment (Zizzamia) [inline site/docs/components/TransactionWrapper.tsx:50]: @abcrane123 Let's revisit in the next PR, what's our `console.log` dev experience strategy.

File: site/docs/pages/transaction/transaction.mdx [hunk 6]
```diff
@@ -141,10 +143,21 @@ export default function TransactionComponents() {
...
-    {({ address, contracts, onError }) => {
+    {({ address, contracts, onError, onSuccess }) => {
       if (address) {
         return (
-          <Transaction address={address} contracts={contracts} onError={onError}>
+          <Transaction
+            address={address}
+            capabilities={{
+              paymasterService: {
+                url: 'https://api.developer.coinbase.com/rpc/v1/base-sepolia/KkgONfLFgUZKI8nnrttLbI78U2vyhtXg',
```
Comment (Zizzamia) [inline site/docs/pages/transaction/transaction.mdx:153]: @abcrane123 Let's reflect in the next PR our strategy in sharing those Key.

File: site/docs/pages/transaction/types.mdx (hunk 2 above)
Comment (Zizzamia) [reply site/docs/pages/transaction/types.mdx:77]: what I meant, is that when we document a type, we should also export it https://github.com/coinbase/onchainkit/blob/main/src/transaction/index.ts :)

**Review by Zizzamia — APPROVED** [review]

**Pull request merged**